### DISCUSSION TITLE
getRSSI update speed limit?

### DISCUSSION THREAD

**Discussion by JustAFriendlyCat** [jgromes/RadioLib#744] [Q&A]
Hello, I'm trying to make a spectrum analyzer and I want it to update pretty fast.

The loop looks like this:

```
for (uint16_t i = 0; i < 200; i++) {
	radio.setFrequency(frequency);
	radio.receiveDirectAsync();
	vTaskDelay(pdMS_TO_TICKS(2));
	rssiData[i] = radio.getRSSI();
	frequency += 0.1f;
}
frequency = 418.00f;
```
For some reason when the delay is below 5ms, it becomes unreliable, sometimes it just hangs, sometimes it kinda works but only for the first few 100 data points.

I'm hoping someone can provide some insight as to why this happens.

**Comment by JustAFriendlyCat** [comment]
I forgot to add I'm using a CC1101 module on a ESP32S3.

**Reply by jgromes** [reply]
I don't think you will be able to create a usable spectrum analyzer from CC1101. You will always see just one point in the time-frequency waterfall. Also, I don't know if the RSSI measurement on CC1101 can be configured e.g. to only measure in some bandwidth, or even how long it measures for, what sort of filtering/averaging it uses etc.

To your question, I don't know why it's behaving like that, though it's likely this is way beyond its intended usage.
